modules/chatops4s-slack/src/main/scala/chatops4s/slack/SlackServer.scala

```diff
@@ -0,0 +1,97 @@
+package chatops4s.slack
+
+import cats.effect.{IO, Resource}
+import cats.implicits.*
+import chatops4s.slack.models.*
+import chatops4s.slack.models.SlackModels.given
+import com.comcast.ip4s.*
+import io.circe.parser.*
+import org.http4s.ember.server.EmberServerBuilder
+import org.http4s.server.Server
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import sttp.tapir.*
+import sttp.tapir.json.circe.*
+import sttp.tapir.server.http4s.Http4sServerInterpreter
+import sttp.tapir.swagger.bundle.SwaggerInterpreter
+import sttp.tapir.openapi.circe.yaml.*
+
+import java.net.URLDecoder
+import java.nio.charset.StandardCharsets
+
+class SlackServer(
+                   config: SlackConfig,
+                   inboundGateway: SlackInboundGateway
+                 ) {
+
+  implicit val logger: Logger[IO] = Slf4jLogger.getLogger[IO]
+
+  private val interactionsEndpoint = endpoint
+    .post
+    .in("slack" / "interactions")
+    .in(formBody[String])
+    .out(stringBody)
+    .name("slack-interactions")
+    .description("Handle Slack interactive button clicks")
+    .tag("Slack")
+    .serverLogicSuccess[IO] { payload =>
+      handleInteraction(payload).as("OK")
+    }
+
+  private val healthEndpoint = endpoint
+    .get
+    .in("health")
+    .out(stringBody)
+    .name("health-check")
+    .description("Health check endpoint")
+    .tag("System")
+    .serverLogicSuccess[IO](_ => IO.pure("OK"))
+
+  private val apiEndpoints = List(interactionsEndpoint, healthEndpoint)
+
+  // Generate OpenAPI docs
+  private val openApiDocs = SwaggerInterpreter()
```

Comment:
we dont need the docs as we are implementing pre-defined api. Thsi can be dropped.